datafusion/sqllogictest/test_files/group_by.slt

```diff
@@ -2129,41 +2129,11 @@ query III
 1 2 1550
 1 3 2175
 
-
-# test_source_sorted_groupby2
-# If ordering is not important for the aggregation function, we should ignore the ordering requirement. Hence
-# "ORDER BY a DESC" should have no effect.
-query TT
-EXPLAIN SELECT a, d,
- SUM(c ORDER BY a DESC) as summation1
- FROM annotated_data_infinite2
- GROUP BY d, a
-----
-logical_plan
-Projection: annotated_data_infinite2.a, annotated_data_infinite2.d, SUM(annotated_data_infinite2.c) ORDER BY [annotated_data_infinite2.a DESC NULLS FIRST] AS summation1
---Aggregate: groupBy=[[annotated_data_infinite2.d, annotated_data_infinite2.a]], aggr=[[SUM(CAST(annotated_data_infinite2.c AS Int64)) ORDER BY [annotated_data_infinite2.a DESC NULLS FIRST]]]
-----TableScan: annotated_data_infinite2 projection=[a, c, d]
-physical_plan
-ProjectionExec: expr=[a@1 as a, d@0 as d, SUM(annotated_data_infinite2.c) ORDER BY [annotated_data_infinite2.a DESC NULLS FIRST]@2 as summation1]
---AggregateExec: mode=Single, gby=[d@2 as d, a@0 as a], aggr=[SUM(annotated_data_infinite2.c)], ordering_mode=PartiallySorted([1])
-----StreamingTableExec: partition_sizes=1, projection=[a, c, d], infinite_source=true, output_ordering=[a@0 ASC NULLS LAST]
-
-query III
+statement error DataFusion error: This feature is not implemented: ORDER BY is not implemented for SUM
```

Comment:
I am not sure if I understand the requirement of https://github.com/apache/arrow-datafusion/issues/9924 correctly:

In my PR, I have `support_ordering` to true for `first`, `last`, `nth_value` and `array_agg_ordered`. For all the other aggregate functions, `support_ordering` is false and `ORDER BY` returns `not implemented` error.

Is this what we want? This seems to be a breaking change for me.

Reply:
I think we should rewrite the test that excluded ordering for SUM in this case 🤔

Reply:
@mustafasrepo  / @ozankabak  / @metesynnada  can you help us understand if the ordering for `SUM(c ORDER BY a DESC)` has meaning?

DataFusion currently ignores cases

We propose making it an error to include clauses that make no sense

However, I could see an argument for permitting the user to write `ORDER BY` even if the aggregate didn't care about ordering (and simply remove the ORDER BY as a optimization) 

I can also see the rationale for failing fast and erroring as the user probably didn't meant to specify an ordering on SUM 🤔

Reply:
This depends on the data type. For integral types, `ORDER BY` doesn't matter for `SUM` (but it shouldn't be an error to still specify it -- the optimizer should just remove it). For floating-point types, it actually makes a difference in the result.

Reply:
I hadn't thought about the implications for `SUM(ORDER BY ...)` for floats 🤔  It appears that this is consistent with what postgres does:


```sql
postgres=# create table foo (x float);
CREATE TABLE
postgres=# insert into foo values (1.0);
INSERT 0 1
postgres=# insert into foo values (2.0);
INSERT 0 1
postgres=# insert into foo values (-1.0);
INSERT 0 1
postgres=# select sum(x ORDER BY x) from foo;
 sum
-----
   2
(1 row)

postgres=# select sum(x IGNORE NULLS) from foo;
ERROR:  syntax error at or near "IGNORE"
LINE 1: select sum(x IGNORE NULLS) from foo;
                     ^
postgres=#
```

I also verified that postgres actually does sort the input:
```sql
postgres=# explain select sum(x ORDER BY x) from foo;
                            QUERY PLAN
-------------------------------------------------------------------
 Aggregate  (cost=169.81..169.82 rows=1 width=8)
   ->  Sort  (cost=158.51..164.16 rows=2260 width=8)
         Sort Key: x
         ->  Seq Scan on foo  (cost=0.00..32.60 rows=2260 width=8)
(4 rows)

postgres=#
```